gef.py

```diff
@@ -5746,6 +5748,41 @@ def search_pattern_by_address(self, pattern: str, start_address: int, end_addres
             del mem
 
         return locations
+        
+    def search_binpattern_by_address(self, binpattern: bytes, start_address: int, end_address: int) -> List[Tuple[int, int, Optional[str]]]:
+        """Search a binary pattern within a range defined by arguments."""
+
+        step = 0x400 * 0x1000
+        locations = []
+
+        for chunk_addr in range(start_address, end_address, step):
+            if chunk_addr + step > end_address:
+                chunk_size = end_address - chunk_addr
+            else:
+                chunk_size = step
+
+            try:
+                mem = gef.memory.read(chunk_addr, chunk_size)
+            except gdb.error as e:
+                estr = str(e)
+                if estr.startswith("Cannot access memory "):
+                    err(estr)
+                    return []
+                else:
+                    raise e
+            for match in re.finditer(binpattern, mem):
+                start = chunk_addr + match.start()
+                preview = str(mem[slice(*match.span())][0:10]) + "..."
```

Comment:
Same, no hardcoded value (`10`): allow tuning via config or make it a global constant

Reply:
done!